### PR TITLE
codegen will update in repository

### LINKED ISSUE #1730: update-codegen.sh will generate file out of repository
**What happened**:
When we run `hack/update-codegen.sh`, updated files will generated on `GOPATH`, not in repository.

**What you expected to happen**:
files in repository are updated

**How to reproduce it (as minimally and precisely as possible)**:
change types and run`hack/update-codegen.sh`

**Anything else we need to know?**:

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fix file will not updated after run `hack/update-codegen.sh`

**Which issue(s) this PR fixes**:
Fixes #1730

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -8,11 +8,29 @@ set -o pipefail
...
 export GOPATH=$(go env GOPATH | awk -F ':' '{print $1}')
 export PATH=$PATH:$GOPATH/bin
 
+go_path="${REPO_ROOT}/_go"
+cleanup() {
+  rm -rf "${go_path}"
+}
+trap "cleanup" EXIT SIGINT
+
+cleanup
```
Comment (RainbowMango) [inline hack/update-codegen.sh:27]: Why keep this since we already put the `cleanup` in `trap`?

**Comment by RainbowMango** [comment]
That's indeed a relief for guys who don't clone Karmada in `GOPATH`. It always creates a `dummy`/`temp` GOPATH and ignores whether it is in GOPATH already.

File: hack/update-codegen.sh (hunk 1 above)
Comment (ikaven1024) [reply hack/update-codegen.sh:27]: Copy it from: 
https://github.com/karmada-io/karmada/blob/2974563c6a14c9fd4b85bacc8ae44ec6013bf99b/hack/verify-codegen.sh#L15-L20

It's cannot be careful.

Comment (RainbowMango) [reply hack/update-codegen.sh:27]: Can you help confirm it? I don't think `cleanup` is needed here.

Comment (ikaven1024) [reply hack/update-codegen.sh:27]: `go_path` maybe contaminated by other programs/scripts, it is safe to clean it before generate files.

Comment (RainbowMango) [reply hack/update-codegen.sh:27]: Yeah, that makes sense to me.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**